workspaces/util/src/__tests__/chunkBySize-test.ts

```diff
@@ -0,0 +1,78 @@
+import { describe, expect, it } from "@jest/globals";
+import { chunkBySize } from "../chunkBySize";
+
+describe("chunkBySize", () => {
+  it("splits an input array into smaller arrays", () => {
+    const result = chunkBySize([1, 2, 3, 4], 2);
+    expect([...result]).toStrictEqual([
+      [1, 2],
+      [3, 4],
+    ]);
+  });
+
+  it("handles an array not perfectly divisible by chunk size", () => {
+    const result = chunkBySize([1, 2, 3], 2);
+    expect([...result]).toStrictEqual([[1, 2], [3]]);
+  });
+
+  it("handles a chunk size greater than array length", () => {
+    const result = chunkBySize([1, 2], 3);
+    expect([...result]).toStrictEqual([[1, 2]]);
+  });
+
+  it("handles a chunk size of 1", () => {
+    const result = chunkBySize([1, 2, 3], 1);
+    expect([...result]).toStrictEqual([[1], [2], [3]]);
+  });
+
+  it("handles an empty array", () => {
+    const result = chunkBySize([], 2);
+    expect([...result]).toStrictEqual([]);
+  });
+
+  it("handles non-numeric elements", () => {
+    const result = chunkBySize(["a", "b", "c"], 2);
+    expect([...result]).toStrictEqual([["a", "b"], ["c"]]);
+  });
+
+  it("throws an error for a chunk size less than 1", () => {
```

Comment:
You could use `it.each` if you wanted to test multiple values with less repetition. (I think I saw -1 in an earlier version of this?) Feel free to search for other uses of `it.each` in this codebase if you want to see some examples.